libr/anal/base_types.h

```diff
@@ -0,0 +1,17 @@
+#ifndef R_BASE_TYPES_H
+#define R_BASE_TYPES_H
+
+#include "r_util.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+R_IPI void enum_type_fini(void *e, void *user);
```

Comment:
why is user needed? the object going to be freed should contain all the pointers inside. also fini != free

Reply:
Yeah I remember inspiring from a function that named it fini to use as a RVectorFree so it stuck with me, I'll rename it, User is needed because it's part of the signature for RVectorFree ```typedef void (*RVectorFree)(void *e, void *user);```